instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/SnippetInjectingResponseWrapper.java

```diff
@@ -20,13 +20,23 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpServletResponseWrapper;
 
+/**
+ * Notes with contentLength: 1. Q: what is the timing that we would update the content length? 1.
+ * the injection happened 2. the content length was already set outside 2. Q: Why we would not add
+ * the content length after the injection if the content length was not set yet? A: The server may
+ * monitor the bytes that have written, and use the new length itself
```

Comment:
```suggestion
 * Notes on Content-Length: the snippet length is only added to the content length when injection occurs and the content length was set previously.
 *
 * If the Content-Length is set after snippet injection occurs (either for the first time or is set again for some reason),
 * we intentionally do not add the snippet length, because the application server may be making that call at the end
 * of a request when it sees the request has not been submitted, in which case it is likely using the real length of
 * content that has been written, including the snippet length.
```